doc/md/writing-motoko/async-star.md

```diff
@@ -0,0 +1,181 @@
+Similar to how `await` is used to consume the result of a future, `await*` is used to produce the result of a computation (by demanding another execution of its body).
+
+From a typing perspective, futures and computations are very similar. Where they differ is in their dynamic behavior: a future is a stateful object that holds the result of a scheduled, asynchronous task while a computation is just an inert value describing a task.
+
+Unlike `await` on a future, `await*` on a computation does not suspend the awaiter, it just immediately executes the computation much like an ordinary function call.
+This means that awaiting an `async*` value only suspends its execution (to complete asynchronously), if the body of the `async*` does a proper `await`.
+The `*` on these expressions is meant to indicate that the computation may involve 0 or more ordinary `await` expressions, and thus may be interleaved with the execution of other messages.
+
+To create an `async*` value, you can just use an `async*` expression, but more typically, you'll declare a local function that returns an `async*` type.
+
+To compute the result of an `async*` computation, you just use an `await*`.
+
+``` motoko
+actor class (Logger : actor { log : Text -> async () }) {
+
+  var logging = true;
+
+  func maybeLog(msg : Text) : async* () {
+    if (logging) { await Logger.log(msg) };
+  };
+
+  func doStuff() : async () {
+    // do stuff
+    await* maybeLog("stuff");
+    // do more stuff
+    await* maybeLog("more stuff");
+  }
+}
+```
+
+One notable difference between `async` and `async*` expressions is that the former are eager, while the latter are not.
+This means that calling the async version of `maybeLog` will eagerly schedule its body to run, even if the `async` result (a future) of the call is never `await`ed.
+Awaiting the same future another time will always produce the original result: the message is executed just once.
+
+On the other hand, calling the `async*` version of `maybeLog` will do nothing unless the result is `await*`-ed, and `await*`-ing the same computation several times will repeat
+the computation each time.
+
+For another example, suppose we define a
+clap function with the side-effect of printing "clap".
+
+``` motoko name=clap
+import Debug "mo:base/Debug"
+func clap() { Debug.print("clap") }
+```
+
+Now, using futures, the code will clap once:
+
+``` motoko include=clap
+let future = async { clap() };
+```
+
+This remains the case, no matter how often you await `future`.
+For example:
+
+``` motoko include=clap
+let future = async { clap() };
+await future;
+await future;
+```
+
+Using computations, on the other hand, the following definition has no effect on its own:
+
+``` motoko include=clap
+let computation = async* { clap() };
+```
+But, the following example will clap twice:
+
+``` motoko include=clap
+let computation = async* { clap() };
+await* computation;
+await* computation;
+```
+
+
```

Comment:
Maybe it should be mentioned that both `async` and `async*` require send capabilities statically.

Reply:
I guess the language manual mentions that.